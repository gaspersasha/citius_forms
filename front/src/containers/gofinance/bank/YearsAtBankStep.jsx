import React from 'react';
import PropTypes from 'prop-types';
import { InputSelect } from '~components';
import s from './bank.module.sass';
import financeStyles from '../styles/finance.module.sass';

const YearsAtBankStep = ({
  onChange,
  yearsAtBank,
  yearsAtBankOptions,
  monthAtBank,
  monthAtBankOptions,
  monthAtBankIsDisabled,
}) => (
  <div className={s.yearsFormWrapper}>
    <div className={s.yearsInputWrapper}>
      <InputSelect
        className={financeStyles.inputSelect}
        label="How many years have you been a member of this bank?"
        subTitle="Please provide how many years you have had an account with this bank"
        value={yearsAtBank.value}
        status={yearsAtBank.status}
        options={yearsAtBankOptions}
        onChange={onChange}
        data-id="How many years have you been a member of this bank?"
        name="yearsAtBank"
      />
    </div>
    <div>
      <InputSelect
        className={financeStyles.inputSelect}
        disabled={monthAtBankIsDisabled}
        label="How many months have you been a member of this bank?"
        subTitle="Please provide how many months you have had an account with this bank"
        status={monthAtBank.status}
        value={monthAtBank.value}
        placeholder="-- Months at bank --"
        options={monthAtBankOptions}
        onChange={onChange}
        data-id="How many months have you been a member of this bank?"
        name="monthAtBank"
      />
    </div>
  </div>
);

YearsAtBankStep.propTypes = {
  onChange: PropTypes.func.isRequired,
  yearsAtBank: PropTypes.shape({
    value: PropTypes.string,
    status: PropTypes.string,
  }).isRequired,
  yearsAtBankOptions: PropTypes.array.isRequired,
  monthAtBank: PropTypes.shape({
    value: PropTypes.string,
    status: PropTypes.string,
  }).isRequired,
  monthAtBankOptions: PropTypes.array.isRequired,
  monthAtBankIsDisabled: PropTypes.bool.isRequired,
};

export default YearsAtBankStep;
