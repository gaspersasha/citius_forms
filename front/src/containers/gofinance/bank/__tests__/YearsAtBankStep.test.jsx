import React from 'react';
import { INPUT_STATUS } from '~constants';
import YearsAtBankStep from '../YearsAtBankStep';

describe('<YearsAtBankStep />', () => {
  const props = {
    onChange: jest.fn(),
    yearsAtBank: {
      value: '2',
      status: INPUT_STATUS.VALID,
    },
    yearsAtBankOptions: [
      { option: '0 Year(s)', value: '0' },
      { option: '1 Year(s)', value: '1' },
      { option: '2 Year(s)', value: '2' },
      { option: '3 Year(s)', value: '3' },
      { option: '4+ Year(s)', value: '4' },
    ],
    monthAtBank: {
      value: '3',
      status: INPUT_STATUS.VALID,
    },
    monthAtBankOptions: [
      { option: '0 Month(s)', value: '0' },
      { option: '1 Month(s)', value: '1' },
      { option: '2 Month(s)', value: '2' },
      { option: '3 Month(s)', value: '3' },
      { option: '4 Month(s)', value: '4' },
      { option: '5 Month(s)', value: '5' },
      { option: '6 Month(s)', value: '6' },
      { option: '7 Month(s)', value: '7' },
      { option: '8 Month(s)', value: '8' },
      { option: '9 Month(s)', value: '9' },
      { option: '10 Month(s)', value: '10' },
      { option: '11 Month(s)', value: '11' },
    ],
    monthAtBankIsDisabled: false,
  };

  const wrapper = shallow(<YearsAtBankStep {...props} />);

  it('renders without errors', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
