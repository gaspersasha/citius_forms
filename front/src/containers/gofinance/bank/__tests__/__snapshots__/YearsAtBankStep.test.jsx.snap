// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<YearsAtBankStep /> renders without errors 1`] = `
<div
  className="yearsFormWrapper"
>
  <div
    className="yearsInputWrapper"
  >
    <InputSelect
      className="inputSelect"
      data-id="How many years have you been a member of this bank?"
      label="How many years have you been a member of this bank?"
      name="yearsAtBank"
      onChange={[MockFunction]}
      options={
        Array [
          Object {
            "option": "0 Year(s)",
            "value": "0",
          },
          Object {
            "option": "1 Year(s)",
            "value": "1",
          },
          Object {
            "option": "2 Year(s)",
            "value": "2",
          },
          Object {
            "option": "3 Year(s)",
            "value": "3",
          },
          Object {
            "option": "4+ Year(s)",
            "value": "4",
          },
        ]
      }
      placeholder=""
      required={false}
      status="valid"
      subTitle="Please provide how many years you have had an account with this bank"
      value="2"
    />
  </div>
  <div>
    <InputSelect
      className="inputSelect"
      data-id="How many months have you been a member of this bank?"
      disabled={false}
      label="How many months have you been a member of this bank?"
      name="monthAtBank"
      onChange={[MockFunction]}
      options={
        Array [
          Object {
            "option": "0 Month(s)",
            "value": "0",
          },
          Object {
            "option": "1 Month(s)",
            "value": "1",
          },
          Object {
            "option": "2 Month(s)",
            "value": "2",
          },
          Object {
            "option": "3 Month(s)",
            "value": "3",
          },
          Object {
            "option": "4 Month(s)",
            "value": "4",
          },
          Object {
            "option": "5 Month(s)",
            "value": "5",
          },
          Object {
            "option": "6 Month(s)",
            "value": "6",
          },
          Object {
            "option": "7 Month(s)",
            "value": "7",
          },
          Object {
            "option": "8 Month(s)",
            "value": "8",
          },
          Object {
            "option": "9 Month(s)",
            "value": "9",
          },
          Object {
            "option": "10 Month(s)",
            "value": "10",
          },
          Object {
            "option": "11 Month(s)",
            "value": "11",
          },
        ]
      }
      placeholder="-- Months at bank --"
      required={false}
      status="valid"
      subTitle="Please provide how many months you have had an account with this bank"
      value="3"
    />
  </div>
</div>
`;
